doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,808 @@
+
+### 2.1.2 Dynamic configuration of mclag_interface
+- In the current implementation the mclag_interfaces configuration is generated from config_db.json file no CLI support is present, a new configuration command is introduced to dynamically add/delete the mclag_interfaces.
+
+### 2.1.3 Keep-alive timer configuration
+- In current implementation the keep-alive timer is hardcoded to 1 second. A new CLI command will be introduced to allow the configuration range of values from 1 to 60 seconds for keep-alive timer.
+
+### 2.1.4 Session timeout configuration
+- In current implementation the session time out 15 seconds after the last keep-alive messages exchange, the timeout duration is hardcoded. A new CLI command will be added to allow a configurable duration for session timeout with the supported range of 3 to 3600 seconds. Session timeout has to be configured in multiples keep-alive timer value. 
+
+### 2.1.5 Static MAC support
+- Current implementation does not support static MAC address over MCLAG. Static MAC information is not exchanged in the FDB TLV between the MCLAG peer nodes. Changes will be added to support Static MAC with TLV.
+- Changes will be added to support the static MAC syncing from local MCLAG node to peer MCLAG node. 
+- Changes will be added to handle the MAC move scenarios in presence of remote Static MAC.
+
+### 2.1.6 Aging disable for ICCP learned MAC addresses
+- In the current implementation, MAC addresses learned in the control plane from the MCLAG peer are installed as dynamic entries. Therefore, if no local traffic from these MAC addresses is seen for the ageing period then they will age out, causing a notification back to ICCPd. However if the mac is learned from peer then it will immediately re-program MAC . The result is transient flooding for the MAC and some unnecessary control plane overhead.
+- When the local MCLAG node learns the MAC addresses from peer MCLAG node the type of the MAC set as dynamic. For the remote MAC addresses learned via ICCP if no traffic received on local MCLAG interface MAC addresses get age out from HW. ICCPd process the age notifications as the MAC address is peer learned re-installs MAC address back to HW.  Transient traffic flooding can occur during remote MAC re-installation. 
+- The process of remote MAC aging and re-installation is repetitive causing un-necessary messaging between modules and processing. 
+- To suppress the unwanted MAC age events, the new implementation programs remote MAC addresses with aging disabled. For the MAC address learned from ICCPd, FdbOrch to set new SAI attribute SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE while programming, which causes SAI to not age out the MAC, but allows it to move.
```

Comment:
I am assuming that this is different for Remote MACs learnt from Orphan ports?

Reply:
@prvattem: Use of the new SAI attribute is applicable to MACs learned from remote orphan ports as well, this is to avoid the periodic aging of MAC addresses.